wireapp/wire-android
# feat: Adding option on bottom sheet to archive use cases

----
#### PR Submission Checklist for internal contributors

- The **PR Title**

- The **PR Description**
----

# What's new in this PR?

### Issues

We need an option to be able to archive conversations visually and consume `updateConversationArchivedStatusUseCase`.

### Solutions

Added the option on the different bottom sheets displayed when interacting with conversations options in `HomeScreen`, `GroupConversationDetailsScreen`, and `OtherUserProfileScreen`. I also added the respective snackbar messages on success and failed states.

### Dependencies (Optional)

- https://github.com/wireapp/kalium/pull/2082

Needs releases with:


### Testing

#### Test Coverage (Optional)


#### How to Test

Load conversations in `HomeScreen`, long tap on a given one and click on `Move to archive` option. Check the conversation disappears and doesn't get displayed anymore. This behavior can also be replicated on Other users profile screen and on Group details screen.

### Attachments (Optional)


https://github.com/wireapp/wire-android-reloaded/assets/2468164/16095fb5-c9c7-4479-bdca-58d245b1101b

----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.
